src/EditorFeatures/TestUtilities/CodeActions/AbstractCodeActionOrUserDiagnosticTest.cs

```diff
@@ -499,19 +499,41 @@ private static async Task VerifyAgainstWorkspaceDefinitionAsync(string expectedT
                         var root = await doc.GetSyntaxRootAsync();
                         var expectedDocument = expectedProject.Documents.Single(d => d.Name == doc.Name);
                         var expectedRoot = await expectedDocument.GetSyntaxRootAsync();
+                        VerifyExpectedDocumentText(expectedRoot.ToFullString(), root.ToFullString());
+                    }
 
-                        var expected = expectedRoot.ToFullString();
-                        if (expected == "")
-                        {
-                            Assert.Equal((object)expected, root.ToFullString());
-                        }
-                        else
-                        {
-                            Assert.Equal(expected, root.ToFullString());
-                        }
+                    foreach (var additionalDoc in project.AdditionalDocuments)
+                    {
+                        var root = await additionalDoc.GetTextAsync();
+                        var expectedDocument = expectedProject.AdditionalDocuments.Single(d => d.Name == additionalDoc.Name);
+                        var expectedRoot = await expectedDocument.GetTextAsync();
+                        VerifyExpectedDocumentText(expectedRoot.ToString(), root.ToString());
+                    }
+
+                    foreach (var analyzerConfigDoc in project.AnalyzerConfigDocuments)
+                    {
+                        var root = await analyzerConfigDoc.GetTextAsync();
+                        var expectedDocument = expectedProject.AnalyzerConfigDocuments.Single(d => d.FilePath == analyzerConfigDoc.FilePath);
+                        var expectedRoot = await expectedDocument.GetTextAsync();
+                        VerifyExpectedDocumentText(expectedRoot.ToString(), root.ToString());
                     }
                 }
             }
+
+            return;
+
+            // Local functions.
+            void VerifyExpectedDocumentText(string expected, string actual)
+            {
+                if (expected == "")
+                {
+                    Assert.Equal((object)expected, actual);
```

Comment:
i thought i wrote a comment why we have this behavior :)  can we move it over as well?

Reply:
I did not find any comment in the original code, but I have tried to retain the original semantics.